Flow.Launcher/Helper/HotKeyMapper.cs

```diff
@@ -0,0 +1,110 @@
+        }
+
+        private static void SetHotkey(string hotkeyStr, EventHandler<HotkeyEventArgs> action)
+        {
+            var hotkey = new HotkeyModel(hotkeyStr);
+            SetHotkey(hotkey, action);
+        }
+
+        internal static void SetHotkey(HotkeyModel hotkey, EventHandler<HotkeyEventArgs> action)
+        {
+            string hotkeyStr = hotkey.ToString();
+            try
+            {
+                HotkeyManager.Current.AddOrReplace(hotkeyStr, hotkey.CharKey, hotkey.ModifierKeys, action);
+            }
+            catch (Exception)
+            {
+                string errorMsg =
+                    string.Format(InternationalizationManager.Instance.GetTranslation("registerHotkeyFailed"),
+                        hotkeyStr);
+                MessageBox.Show(errorMsg);
+            }
+        }
+
+        internal static void RemoveHotkey(string hotkeyStr)
+        {
+            if (!string.IsNullOrEmpty(hotkeyStr))
+            {
+                HotkeyManager.Current.Remove(hotkeyStr);
+            }
+        }
+
+        internal static void OnHotkey(object sender, HotkeyEventArgs e)
+        {
+            if (!ShouldIgnoreHotkeys())
+            {
+                if (settings.LastQueryMode == LastQueryMode.Empty)
+                {
+                    mainViewModel.ChangeQueryText(string.Empty);
+                }
+                else if (settings.LastQueryMode == LastQueryMode.Preserved)
+                {
+                    mainViewModel.LastQuerySelected = true;
+                }
+                else if (settings.LastQueryMode == LastQueryMode.Selected)
+                {
+                    mainViewModel.LastQuerySelected = false;
+                }
+                else
+                {
+                    throw new ArgumentException($"wrong LastQueryMode: <{settings.LastQueryMode}>");
+                }
+
+                mainViewModel.ToggleFlowLauncher();
+                e.Handled = true;
+            }
+        }
+
+        /// <summary>
+        /// Checks if Flow Launcher should ignore any hotkeys
+        /// </summary>
+        /// <returns></returns>
+        private static bool ShouldIgnoreHotkeys()
+        {
+            //double if to omit calling win32 function
+            if (settings.IgnoreHotkeysOnFullscreen)
+                if (WindowsInteropHelper.IsWindowFullscreen())
+                    return true;
+
+            return false;
+        }
+
+        private static void SetCustomPluginHotkey()
+        {
+            if (settings.CustomPluginHotkeys == null) return;
+            foreach (CustomPluginHotkey hotkey in settings.CustomPluginHotkeys)
+            {
+                SetHotkey(hotkey.Hotkey, (s, e) =>
+                {
+                    if (ShouldIgnoreHotkeys()) return;
+                    mainViewModel.MainWindowVisibility = Visibility.Visible;
+                    mainViewModel.ChangeQueryText(hotkey.ActionKeyword);
```

Comment:
Could you also add `mainViewModel.LastQuerySelected = false;` to here because I do have encountered it is selected sometime.